internal/source/source.go

```diff
@@ -0,0 +1,67 @@
+package source
+
+import (
+	"context"
+	"errors"
+	"io/fs"
+	"sync"
+
+	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
+)
+
+type Source interface {
+	SourceContent(context.Context, *rukpakv1alpha1.Bundle) (*Result, error)
+}
+
+type Result struct {
+	Bundle         fs.FS
+	ResolvedSource *rukpakv1alpha1.BundleSource
+	State          State
+	Message        string
+}
+
+type State string
+
+const (
+	StatePending   State = "Pending"
+	StateUnpacking State = "Unpacking"
+	StateUnpacked  State = "Unpacked"
+)
```

Comment:
Do we need an internal/source/types.go?

Reply:
We _could_ I suppose. I stuffed all the general "applies to all sources" stuff into this file, which is still pretty small. Seemed unnecessary to me to split things out.